lib/datasets/dataset.rb

```diff
@@ -33,9 +33,21 @@ def cache_path
       @cache_path ||= CachePath.new(@metadata.id)
     end
 
-    def download(output_path, url, &block)
-      downloader = Downloader.new(url)
-      downloader.download(output_path, &block)
+    def download(output_path, *urls, &block)
+      urls.each do |url|
+        downloader = Downloader.new(url)
+        downloader.download(output_path, &block)
+        return
+      rescue Net::HTTPClientException => error
+        if urls.last != url
+          message = "site is not available: " +
+                    "#{error.class}: #{error.message}: " +
+                    "Attempting to download from an alternative site"
+          $stderr.puts(message)
+        else
+          raise error
+        end
+      end
```

Comment:
Sorry but can we do this in `Downloader`?
We will extract `Downloader` to https://github.com/red-data-tools/red-datastock for easy to share.

Reply:
Yes. I would like to send a this patch to https://github.com/red-data-tools/red-datastock

Reply:
Ah, we don't need to change https://github.com/red-data-tools/red-datastock for now.
It's not ready yet.
We may reconstruct it later.

Reply:
Oh, I'm so sorry. I'll try not to send a patch.